apps/emqx_bridge_gcp_pubsub/src/emqx_bridge_gcp_pubsub_consumer_worker.erl

```diff
@@ -0,0 +1,552 @@
+
+-spec get_subscription(pid()) -> {ok, map()} | {error, term()}.
+get_subscription(WorkerPid) ->
+    gen_server:call(WorkerPid, get_subscription, 5_000).
+
+%%-------------------------------------------------------------------------------------------------
+%% `ecpool' health check
+%%-------------------------------------------------------------------------------------------------
+
+-spec health_check(pid()) -> boolean().
+health_check(WorkerPid) ->
+    case optvar:read(?OPTVAR_SUB_OK(WorkerPid), ?HEALTH_CHECK_TIMEOUT) of
+        {ok, _} ->
+            true;
+        timeout ->
+            false
+    end.
+
+%%-------------------------------------------------------------------------------------------------
+%% `emqx_resource' API
+%%-------------------------------------------------------------------------------------------------
+
+connect(Opts0) ->
+    Opts = maps:from_list(Opts0),
+    #{
+        ack_retry_interval := AckRetryInterval,
+        bridge_name := BridgeName,
+        connector_state := ConnectorState,
+        ecpool_worker_id := WorkerId,
+        hookpoint := Hookpoint,
+        instance_id := InstanceId,
+        pull_max_messages := PullMaxMessages,
+        topic_mapping := TopicMapping
+    } = Opts,
+    TopicMappingList = lists:keysort(1, maps:to_list(TopicMapping)),
+    Index = 1 + (WorkerId rem map_size(TopicMapping)),
+    {Topic, MQTTConfig} = lists:nth(Index, TopicMappingList),
+    Config = #{
+        ack_retry_interval => AckRetryInterval,
+        connector_state => ConnectorState,
+        hookpoint => Hookpoint,
+        instance_id => InstanceId,
+        mqtt_config => MQTTConfig,
+        pull_max_messages => PullMaxMessages,
+        topic => Topic,
+        subscription_id => subscription_id(BridgeName, Topic)
+    },
+    start_link(Config).
+
+%%-------------------------------------------------------------------------------------------------
+%% `gen_server' API
+%%-------------------------------------------------------------------------------------------------
+
+-spec init(config()) -> {ok, state(), {continue, ensure_subscription}}.
+init(Config) ->
+    process_flag(trap_exit, true),
+    State = Config#{
+        ack_timer => undefined,
+        async_workers => #{},
+        pending_acks => []
+    },
+    {ok, State, {continue, ensure_subscription}}.
+
+handle_continue(ensure_subscription, State0) ->
+    case ensure_subscription_exists(State0) of
+        ok ->
+            #{instance_id := InstanceId} = State0,
+            ?tp(
+                debug,
+                "gcp_pubsub_consumer_worker_subscription_ready",
+                #{instance_id => InstanceId}
+            ),
+            ?MODULE:pull_async(self()),
+            optvar:set(?OPTVAR_SUB_OK(self()), subscription_ok),
+            {noreply, State0};
+        error ->
+            %% FIXME: add delay if topic does not exist?!
```

Comment:
will the worker be responsible while topic doesn't exist (or the process will keep looping here and ignoring external messages)?

Reply:
I think we should treat an nonexistent topic as in your PR: https://github.com/emqx/emqx/pull/11090/files#diff-e339f277c61f643708ab4199f7cc1d3af03beb7772ff6fcd0914d5c3e4ea20b9R105